File: lessons/06-parameters/cpp/number_publisher_with_param.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;

class NumberPublisher: public rclcpp::Node 
{
    public:
        NumberPublisher(): Node("number_publisher")
        {
            this->declare_parameter("number",2);//initialize parameter with name 'number' and default value 2
            this->declare_parameter("timer_period",1.0);//initialize parameter with name 'time period' and default value 1.0
            number_=this->get_parameter("number").as_int();//get value of the parameter as an integer
            timer_period_=this->get_parameter("timer_period").as_double();//get value of the parameter as a double

            number_publisher_=this->create_publisher<example_interfaces::msg::Int64>("number",10);
            number_timer_=this->create_wall_timer(std::chrono::duration<double>(timer_period_),
                                                    std::bind(&NumberPublisher::publishNumber,this));
            RCLCPP_INFO(this->get_logger(),"Number publisher node has been started");

        }
    private:
        int number_;
        double timer_period_;
        rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
        rclcpp::TimerBase::SharedPtr number_timer_;
        void publishNumber() {
            auto msg = example_interfaces::msg::Int64();
            msg.data=number_;
            number_publisher_->publish(msg);

        }

};

int main(int argc,char **argv) {
    rclcpp::init(argc,argv);
    auto node=std::make_shared<NumberPublisher>();
    rclcpp::spin(node);
    RCLCPP_INFO(node->get_logger(),"Shutting down node....");
    rclcpp::shutdown();
    return 0;
}

File: lessons/06-parameters/python/number_publisher_with_param.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int32

class NumberPublisher(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.declare_parameter("number",2) #Declare a parameter named 'number' with default value 2
        self.declare_parameter("timer_period",1.0) #Declare a parameter with name 'timer_period', having default value 1s
        self.number_=self.get_parameter("number").value #Get value passed into parameter
        self.timer_period_=self.get_parameter("timer_period").value
        self.publisher_=self.create_publisher(Int32,"number",10)
        self.timer_=self.create_timer(self.timer_period_,self.publish_number)
        self.get_logger().info("Number Publisher has been started.")


    def publish_number(self):
        msg=Int32()
        msg.data=self.number_
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node=NumberPublisher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        print(f"\nShutting down node....")
    finally:
        node.destroy_node()
        if rclpy.ok(): #returns false if ros has already shut down
            rclpy.shutdown()

if __name__=="__main__":
    main()